active-rfcs/0000-scoped-css-with-preserved-specificity.md

```diff
@@ -0,0 +1,146 @@
+- Start Date: 2022-01-04
+- Reference Issues: [#1909](https://github.com/withastro/astro/issues/1909), [#1762](https://github.com/withastro/astro/issues/1762)
+- Implementation PR: 
+
+
+
+# Summary
+
+Preserve specificity in scoped CSS, consistent with the specificity in global CSS.
+
+Astro currently adds an additional class to all elements and selectors, increasing specificity from `O+1` to `O(1)+1`.
+
+Astro can prevent this specificity increase by wrapping the scoping mechanism in a `:where()` CSS pseudo-class.
+
+
+
+# Example
+
+```astro
+<style>
+p {
+  color: blueviolet;
+}
+
+aside p {
+  color: darkviolet;
+}
+
+:global(aside) p {
+  text-decoration: underline;
+}
+</style>
+<p>This is a paragraph.</p>
+<aside>
+  <p>This is a paragraph in an aside.</p>
+</aside>
+```
+
+In this example, the HTML and CSS would generally transform into the following code:
+
+```astro
+<style>
+p:where(.astro-OZV3B5RX) {
+  color: blueviolet;
+}
+
+aside:where(.astro-OZV3B5RX) p:where(.astro-OZV3B5RX) {
+  color: darkviolet;
+}
+
+aside p:where(.astro-OZV3B5RX) {
+  text-decoration: underline;
+}
+</style>
+<p class="OZV3B5RX">This is a paragraph.</p>
+<aside class="OZV3B5RX">
+  <p class="OZV3B5RX">This is a paragraph in an aside.</p>
+</aside>
+```
+
+| Authored Selector  | Compiled Selector                                                                   | Authored Specificity | Compiled Specificity |
+|:------------------ |:----------------------------------------------------------------------------------- |:--------------------:|:--------------------:|
+| `p`                | `p:where(.astro-OZV3B5RX)`                                            | `0.0.1`              | `0.0.1`              |
+| `aside p`          | `aside:where(.astro-OZV3B5RX) p:where(.astro-OZV3B5RX)` | `0.0.2`              | `0.0.2`              |
+| `:global(aside) p` | `aside p:where(.astro-OZV3B5RX)`                                      | `0.0.2`              | `0.0.2`              |
+
+
+By using `:where()`, styles will now:
+
+- respect the specificity with which they were authored.
+- work consistently alongside other CSS files or CSS libraries.
+- still preserve the exclusive boundaries that prevent styles from applying outside the component.
+
+
+
+# Motivation
+
+Astro component styles are scoped by default, meaning they only apply to items in the component.
+These are _exclusive_ styling boundaries that prevent styles from applying to items outside the component or within child components.
+
+This specificity increase is perceptually inconsistent to authors, without a deeper knowledge of the Astro internals and **Type**, **ID**, add **Class** selector specificity.
+This specificity increase makes it hard to combine with other CSS files or other styling libraries (Tailwind, CSS Modules, Styled Components, Stitches), as those systems would not include our additional side-effects.
```

Comment:
Can you provide some examples or where you might use those types of libraries in conjunction with scoped Astro CSS?  For example, I can definitely see where you are getting at with Tailwind, but within an example its unclear to me why you would expect Tailwind's CSS to take precedent.